### ISSUE TITLE
[Bug]: Screen reader is announcing the incorrect position of row numbers for the cells in the table in Windows.

### ISSUE THREAD

**Issue by msftedad** [ceolter/ag-grid#10410]
### Link to reproducible scenario

https://www.ag-grid.com/react-data-grid/row-sorting/

### Describe the bug

Steps to reproduce 

1. Open URL: [index.tsx - nodebox - CodeSandbox](https://codesandbox.io/p/sandbox/krgjhr?file=%2fsrc%2findex.tsx) on latest Edge browser. 
2. Table will appear.
3. Press tab key to reach to the 'Athlete' column header of table.
4. Verify whether screen reader is announcing the correct position of row numbers for the cells in the table or not.

Actual behaviour
On pressing tab, when focus moves to the 'Athlete column header it announces it as 'Athlete column header row 2 column 3'.

https://github.com/user-attachments/assets/187ac1c3-b743-4e81-b132-28dc9acd2ec2

<img width="892" alt="Image" src="https://github.com/user-attachments/assets/d855552b-a009-4c59-9db7-3d53c99a8929" />

Expected behaviour
On pressing tab, when focus moves to the 'Athlete column header it should announce it as 'Athlete column header row 1 column 3'.

More information
Title: Screen reader is announcing the incorrect position of row numbers for the cells in the table in Windows.
User Impact: Low vision users are not able to identify the currently focused control if focus indicator is not visible on receiving focus

### Version

30.1.0

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers

**Comment by ava-utting** [comment]
Hi @msftedad ,

This is a known accessibility limitation in that pinned columns aren’t rendered in sequence with un-pinned columns. This is throwing off the screen readers which expect that elements that are visually adjacent are adjacent in the DOM structure.
Therefore, incorrect announcements are made when you navigate into pinned row/column cells.

This limitation is documented here: https://www.ag-grid.com/vue-data-grid/accessibility/#limitations-in-announcing-the-correct-column-name-in-grouped-columns

Kind Regards,
Ava
